src/Actionbar/Sort/sort.js

```diff
@@ -52,41 +71,45 @@ export default class ActionbarSort extends Component {
       <Popup
         isOpen={this.store.menuOpen}
         trigger={
-          <Button
-            icon={<SortIcon />}
-            onClick={this.store.handleMenuOpen}
-          />
+          <div ref='button'>
```

Comment:
Couldn't put the `ref` on `<List>` and `<Button>` as they are stateless function components. I wonder if there's a better way to listen to a click outside of the button and popup.

I tried simplifying the condition to `this.store.menuOpen` in `handleDocumentClick` and using `event.stopPropagation` in the onClick handlers or `<Button>` and `<List>` but this somehow doesn't prevent handleDocumentClick from being fired. I imagine it might have to do with React's way of registering and handling events.

Reply:
If I understood correctly, you want the popup to close if you click outside the <Button> and <List>, is that correct?

I think [SUI's Popup](https://react.semantic-ui.com/modules/popup#popup-example-click) already handles this case. Maybe try passing `onClose={this.store.handleMenuClose()}` to the popup (if you want a controlled component) or remove `isOpen={this.store.menuOpen}` (if you want an uncontrolled component).

Generally with React we avoid playing with the DOM directly (like `window.document.addEventListener`), but do everything via React's virtual dom.